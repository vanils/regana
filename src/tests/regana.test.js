
const regana = require('../regana.js');
const path = require('path');

describe('regana', () => {
  describe('#analyse', () => {

    test('should expose proper properties', () => {
      expect(regana.analyse).toBeDefined();
    });

    test('analyse should throw error with invalid input', () => {
      expect(() => {
        regana.analyse();
      }).toThrowErrorMatchingSnapshot();
      expect(() => {
        regana.analyse(null);
      }).toThrowErrorMatchingSnapshot();
      expect(() => {
        regana.analyse(NaN);
      }).toThrowErrorMatchingSnapshot();
      expect(() => {
        regana.analyse('');
      }).toThrowErrorMatchingSnapshot();
      expect(() => {
        regana.analyse({});
      }).toThrowErrorMatchingSnapshot();
      expect(() => {
        regana.analyse([]);
      }).toThrowErrorMatchingSnapshot();
    });

    test('analyse should work with valid input', () => {
      expect(regana.analyse(path.resolve(__dirname, './mocks/file1.js'))).toMatchSnapshot();
      expect(regana.analyse(path.resolve(__dirname, './mocks/file2.js'))).toMatchSnapshot();
    });
  });
});