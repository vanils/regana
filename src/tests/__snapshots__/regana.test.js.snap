// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regana #analyse analyse should throw error with invalid input 1`] = `"missing entry"`;

exports[`regana #analyse analyse should throw error with invalid input 2`] = `"missing entry"`;

exports[`regana #analyse analyse should throw error with invalid input 3`] = `"missing entry"`;

exports[`regana #analyse analyse should throw error with invalid input 4`] = `"missing entry"`;

exports[`regana #analyse analyse should throw error with invalid input 5`] = `"missing entry"`;

exports[`regana #analyse analyse should throw error with invalid input 6`] = `"missing entry"`;

exports[`regana #analyse analyse should work with valid input 1`] = `
"{
  \\"segments\\": {
    \\"VariableDeclaration_41c2c0d071fc05f774e45e78ac0ad5fc\\": {
      \\"start\\": 1,
      \\"end\\": 21,
      \\"pointers\\": [],
      \\"file\\": \\"File_8ba5c586d44c70ca777c0f66b5243b40\\",
      \\"id\\": \\"VariableDeclaration_41c2c0d071fc05f774e45e78ac0ad5fc\\"
    }
  },
  \\"exposes\\": {},
  \\"variables\\": {
    \\"testValue\\": \\"VariableDeclaration_41c2c0d071fc05f774e45e78ac0ad5fc\\"
  }
}"
`;

exports[`regana #analyse analyse should work with valid input 2`] = `
"{
  \\"segments\\": {
    \\"VariableDeclaration_ba10fb9b241b9e39052c6a5d8a451d42\\": {
      \\"start\\": 0,
      \\"end\\": 15,
      \\"pointers\\": [],
      \\"file\\": \\"File_aac47180851be5de7f5b19fa7afa91b7\\",
      \\"id\\": \\"VariableDeclaration_ba10fb9b241b9e39052c6a5d8a451d42\\"
    },
    \\"ExportNamedDeclaration_675ae9dfbe2abcc8183078ae43de38a9\\": {
      \\"start\\": 16,
      \\"end\\": 32,
      \\"pointers\\": [
        \\"VariableDeclaration_ba10fb9b241b9e39052c6a5d8a451d42\\"
      ],
      \\"file\\": \\"File_aac47180851be5de7f5b19fa7afa91b7\\",
      \\"id\\": \\"ExportNamedDeclaration_675ae9dfbe2abcc8183078ae43de38a9\\"
    }
  },
  \\"exposes\\": {
    \\"test\\": \\"ExportNamedDeclaration_675ae9dfbe2abcc8183078ae43de38a9\\"
  },
  \\"variables\\": {
    \\"test\\": \\"VariableDeclaration_ba10fb9b241b9e39052c6a5d8a451d42\\"
  }
}"
`;
